collect_app/src/main/java/org/odk/collect/android/logic/FormController.java

```diff
@@ -399,8 +399,15 @@ private boolean indexIsInFieldList(FormIndex index) {
                 // no group
                 return false;
             }
-            FormEntryCaption grp = captions[captions.length - 2];
-            return groupIsFieldList(grp.getIndex());
+            // If at least one of groups you are inside is field list your index is in field list
+            boolean groupIsFieldList;
+            for (FormEntryCaption caption : captions) {
+                groupIsFieldList = groupIsFieldList(caption.getIndex());
```

Comment:
I suggest inlining `groupIsFieldList`. You’ll remove two lines of code and lose no clarity.